app/optscale_api/orgs_api.py

```diff
@@ -2,76 +2,123 @@
-        :type user_id: string
         :param admin_api_key: the secret admin API key
-        :type admin_api_key: string
-        :return: The organization data or None if there is an error or no organization exists.
+        :return: The organization data or None if there is an error.
+        An empty list if no organization exists
+        :raise:
+            UserAccessTokenError If an error occurs while obtaining the access token.
+            A default Exception if an error occurs accessing the organization
 
+        The Optscale API returns a list or dict like the following one:
+        {
+            "organizations": [
+                {
+                    "deleted_at": 0,
+                    "created_at": 1731919809,
+                    "id": "3e61c772-b78a-4345-b7da-5243b09bfe03",
+                    "name": "MyOrg",
+                    "pool_id": "0bc61f62-f280-4a03-bf3f-446b14994594",
+                    "is_demo": false,
+                    "currency": "USD",
+                    "cleaned_at": 0
+                }
+            ]
+        }
         """
         try:
-            # request user's access token
-            user_access_token = await self.auth_client.obtain_user_auth_token_with_admin_api_key(
-                user_id=user_id,
-                admin_api_key=admin_api_key
-            )
-            if user_access_token is None:
-                logger.warning(TOKEN_ERROR_MESSAGE.format(user_id))
-                return None
             # get the user's org
-            response = await self.api_client.get(endpoint=ORG_ENDPOINT,
-                                                 headers=self.auth_client.build_bearer_token_header(
-                                                     bearer_token=user_access_token))
-            if len(response) == 0:
-                logger.info(NO_ORG_MESSAGE.format(user_id))
-                return {}
-            else:
-                logger.info(ORG_RETRIEVED_MESSAGE.format(user_id))
-                return response
+            user_access_token = await get_user_access_token(
+                user_id=user_id, admin_api_key=admin_api_key, auth_client=auth_client
+            )
+            response = await self.api_client.get(
+                endpoint=ORG_ENDPOINT,
+                headers=build_bearer_token_header(bearer_token=user_access_token),
+            )
+
+            if response.get("error"):
+                logger.error(
+                    f"Failed to get the org data from OptScale for the user : {user_id}"
+                )
+                raise OptScaleAPIResponseError(
+                    title="Error response from OptScale",
+                    reason=response.get("data", {}).get("error", {}).get("reason", ""),
+                    status_code=response.get(
+                        "status_code", http_status.HTTP_403_FORBIDDEN
+                    ),
+                )
```

Comment:
See this is a great example of what I was talking in one of my first comments about the `api_client` methods -- you're effectively parsing the response twice (not a performance issue but it increases the codebase's complexity) and you could have just used `AsyncClient.get` and then the response handling would be more or less the same. Even replacing it with `raise_for_status()` would have worked, albeit with less descriptive error code

However, I still like the idea of the abstraction and it can be used to automate the logging which is super useful.